guides/hack/03-expressions-and-operators/01-introduction.md

```diff
@@ -1,29 +1,43 @@
-**Topics covered in this section**
+When combined, operators evaluate according to their associativity. For more information, see [Operator Precedence](/hack/expressions-and-operators/operator-precedence).
 
-* [Arithmetic, `+`, `*` etc](addition.md)
-* [Assignment, `=`, `+=` etc](assignment.md)
-* [await](await.md)
-* [Bitwise Operators](bitwise-operators.md)
-* [Casting, `(int)`, `(float)`, `(bool)` `(string)`](casting.md)
-* [Coalesce, `??`](coalesce.md)
-* [Comparisons, `>` etc](comparisons.md)
-* [Decrement, `--`](incrementing-and-decrementing.md)
-* [echo intrinsic function](echo.md)
-* [Equality operators](equality.md)
-* [Error control, `@`](error-control.md)
-* [exit intrinsic function](exit.md)
-* [Increment, `++`](incrementing-and-decrementing.md)
-* [invariant intrinsic function](invariant.md)
-* [list intrinsic function](list.md)
-* [Logical operators, `&&`, `||`, `!`](logical-operators.md)
-* [Member selection, `->`, `?->`](member-selection.md)
-* [new](new.md)
-* [Operator precedence](operator-precedence.md)
-* [Pipe, `|>`](pipe.md)
-* [Scope resolution, `::`](scope-resolution.md)
-* [Spaceship, `<=>`](equality.md)
-* [String concatenation, `.`](string-concatenation.md)
-* [Subscripting, `[]`](subscript.md)
-* [Ternary, `?` `:`](ternary.md)
-* [Type Assertions `is`, `as`, `?as`](type-assertions.md)
-* [Yield](yield.md)
+## Assignment Operators
+* [Assignment](/hack/expressions-and-operators/assignment) (`=`, `+=`, and more)
+* [Bitwise](/hack/expressions-and-operators/bitwise-operators) (`&`, `|`, `^`, `<<`, `>>`, `~`)
```

Comment:
The operators in this list are not assignment operators. Did you mean `&=` and such? I do not know for sure which operators can be combined with assignment, so maybe `&=` does not exist.

Reply:
In all of these, I was perhaps using too loose a description of assignment, as in anything that writes or updates a value.

Reply:
I say this because we are "assigning" values in all these cases. 

If I could push back, In `+=` I consider the assignment to be the increment to the value at the variable. In `<<` the shift assigns a different value by shifting, etc. Same for other bitwise. Assign a value based on the result of binary operations.

Reply:
I see where our definitions differ. I don't consider "produces a new value" to be assignment. I consider giving a new value to a name assignment. We both agree that `$x += 5` is an assignment. In don't consider `$x << 5` to be an assignment, because `$x` is not altered. `$y = $x << 5` would be an assignment by my definition, because of the `$y = ...` part.

Hack disallows using assignment as an expression. So if `if (...) 0;` emits a typechecker error like `Assignments can no longer be used as expressions. Pull the assignment out into a separate statement.`, the syntax at the `...` is an assignment. `$x += 5` emits this error, `$x << 5` does not, and `$y = $x << 5` emits the error again. Calling `<<` an assignment may be cause for confusion.